### PR TITLE
Add address into ID methods

### PR DESCRIPTION
**Pull request by thibault-martinez**
To avoid some copies.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: types/src/block/address/nft.rs [hunk 1]
```diff
@@ -25,11 +25,17 @@ impl NftAddress {
         Self::from(id)
     }
 
-    /// Returns the [`NftId`] of an [`NftAddress`].
+    /// Returns the [`NftId`] of a [`NftAddress`].
```
Comment (Thoralf-M) [inline types/src/block/address/nft.rs]: Isn't an correct here?

Comment (samuel-rufi) [reply types/src/block/address/nft.rs]: 🤔

Comment (thibault-martinez) [reply types/src/block/address/nft.rs]: I'm always confused with that, idk

Comment (thibault-martinez) [inline types/src/block/address/nft.rs]:
```suggestion
    /// Returns the [`NftId`] of an [`NftAddress`].
```

File: types/src/block/address/nft.rs [hunk 2]
```diff
@@ -25,11 +25,17 @@ impl NftAddress {
...
     #[inline(always)]
     pub fn nft_id(&self) -> &NftId {
         &self.0
     }
+
+    /// Consumes a [`NftAddress`] and returns its [`NftId`].
```
Comment (thibault-martinez) [inline types/src/block/address/nft.rs]:
```suggestion
    /// Consumes an [`NftAddress`] and returns its [`NftId`].
```

**Review by Thoralf-M — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**